graph/pull.go

```diff
@@ -143,7 +143,10 @@ func (s *TagStore) pullRepository(r *registry.Session, out io.Writer, repoInfo *
 	errors := make(chan error)
 
 	layersDownloaded := false
-	for _, image := range repoData.ImgList {
+	imgIDs := []string{}
+	defer s.graph.Release(imgIDs, imgIDs)
```

Comment:
I think we have the same issue here where this deferred function will be called with an empty slice and not release anything.